Dockerfile

```diff
@@ -1,11 +1,13 @@
-FROM alpine:latest
-RUN mkdir /app
-COPY . /app
+FROM alpine:3.19 as builder
+COPY . /tmp/app
 RUN apk update && \
     apk add build-base cmake gdb valgrind cmocka-dev openssl-dev
-RUN mkdir /app/build
-WORKDIR /app/build
-RUN cmake .. && \
-    cmake --build .
-WORKDIR /app
-CMD ["/app/build/main"]
+
+RUN mkdir /tmp/app/build && \
+    cd /tmp/app/build && \
+    cmake /tmp/app && \
+    cmake --build /tmp/app/build
+
+FROM alpine:3.19
+COPY --from=builder /tmp/app/build/main /
```

Comment:
Part of the reason we have all those tools in the container is so we can run the tests and Valgrind in the pipeline. If there's a failure, we can debug by going into the container and running gdb. How would you suggest accommodating that workflow? Do we have a separate "development" container?

Reply:
The docker build action supports the `--target` directive for docker build meaning that two Containers could be built from the same Dockerfile.

Separating developer container from app container would mean that users would be delivered a container for their use case.